### PR TITLE
run tests from all the drafts, with explicit unsupported list

### PR DESCRIPTION
**Pull request by ChALkeR**
Still investigating which of these need fixes.

### CODE REVIEW

File: formats.js [hunk 1]
```diff
@@ -19,6 +19,7 @@ exports['hostname'] = function(input) {
 
   return true
 }
+exports['host-name'] = exports['hostname'] // draft3 backwards compat
```
Comment (ChALkeR) [inline formats.js]: should be in a separate pr

File: test/json-schema.js [hunk 2]
```diff
@@ -6,40 +6,84 @@ const validator = require('../')
...
+  'if-then-else.json',
+  //  draft2019-09
+  'minContains.json',
+  'maxContains.json',
+  'anchor.json',
+  'dependentSchemas.json',
+  'dependentRequired.json',
+  'unevaluatedProperties.json',
+  'unevaluatedItems.json',
+  'defs.json', // FIXME: fails open!
```
Comment (ChALkeR) [inline test/json-schema.js:28]: This bug should be solved before we can use this, I'm at it.
Tests can land regardless.

**Review by kklash — APPROVED** [review]
tACK
